### PR TITLE
feat: bundle xjst with webpack instead of ready bundle

### PR DESCRIPTION
**Pull request by Guria**

### CODE REVIEW

**Comment by Guria** [comment]
@awinogradov so, what do you say? this pr doesn't introduces my other crazy ideas, but only get's rid of rebundling and unneccessary `postinstall` script.
The only thing you may not like here is that I am pointing to extracted `xjst-vidom` engine. I can point it back to your fork of `bem-xjst` if you wish. But I think it more flexible to have it as separate repo.

**Comment by awinogradov** [comment]
I agree. But arch is wrong. We should call `bem-xjst` for apply, not separated module. Now it's okay. It's a good point to start engines separating.

File: package.json [hunk 1]
```diff
@@ -6,16 +6,16 @@
...
     "predeploy": "cd examples && npm i && npm run build",
-    "deploy": "gh-pages -d examples/dist",
-    "postinstall": "cd ./node_modules/bem-xjst && npm run make && cd ../.. && npm run build"
+    "deploy": "gh-pages -d examples/dist"
   },
   "dependencies": {
     "bem-cn": "^1.2.2",
-    "bem-components": "git+ssh://git@github.com/awinogradov/bem-components.git#v2",
-    "bem-core": "git+ssh://git@github.com/bem/bem-core.git#v2",
+    "bem-components": "github:awinogradov/bem-components#v2",
```
Comment (awinogradov) [inline package.json]: use from npm. `bem-core` and `bem-components` are available.

File: src/provider/provider.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 const reactXjst = require('react-xjst');
 const React = require('react');
-const templates = require('../../dist/templates.js');
+const Runtime = require("xjst-vidom");
```
Comment (awinogradov) [inline src/provider/provider.js]: '

File: webpack.config.js [hunk 3]
```diff
@@ -2,22 +2,15 @@ const path = require('path');
 const bemLoader = require('bem-loader');
 const CollectBemAssetsPlugin = bemLoader.CollectBemAssetsPlugin;
 const fs = require('fs-extra');
-const bemxjst = require('bem-xjst');
+function wrapTemplates (code) {
```
Comment (awinogradov) [inline webpack.config.js]: Why we should do this? @Guria can you explain?

Comment (Guria) [inline webpack.config.js]: of course. it allows not to reevaluate templates in compile call: https://github.com/bem/bem-xjst/blob/master/lib/bemxjst/index.js#L120-L122
see also https://github.com/bem/bem-xjst/pull/285

File: package.json (hunk 1 above)
Comment (Guria) [inline package.json]: should we stick to v2 or it's safe to try newer versions?

File: src/provider/provider.js (hunk 2 above)
Comment (Guria) [inline src/provider/provider.js]: don't get it

File: webpack.config.js (hunk 3 above)
Comment (awinogradov) [inline webpack.config.js]: Cool! Ok, now it's okay

File: package.json (hunk 1 above)
Comment (awinogradov) [inline package.json]: It doesn't depend on version. I hope ;)

**Comment by Guria** [comment]
> We should call bem-xjst for apply, not separated module.

`BEMTREE`, `BEMHTML` and any other engine runtimes is just sub-class of `BEMXJST`. All of them has needed methods. The only need to call `bem-xjst` module is generate browserify bundled runtime + templates on fs.

**Comment by awinogradov** [comment]
let it be

**Comment by Guria** [comment]
smth else?

**Comment by awinogradov** [comment]
I think about engines as plugins. When we can use them all at the same moment.

**Comment by Guria** [comment]
we can live with current "engine as inhereted package" for now

**Comment by Guria** [comment]
So does it looks good for you to be merged for now?

**Comment by awinogradov** [comment]
🔥 We can create `xjst-*` repos in `bem` or `bem-contrib`, I can ;)

**Comment by Guria** [comment]
I also have access to create repos in bem-contrib, can move it at any time. Do you think current state of xjst-vidom worth it?

**Pull request merged**